src/plugins/intel_cpu/src/nodes/unique.cpp

```diff
@@ -0,0 +1,453 @@
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    if (!one_of(op->get_input_size(), 1, 2) || op->get_output_size() != 4)
+        THROW_ERROR << "has incorrect number of input/output edges.";
+
+    for (int i = 0; i < 4; i++) {
+        definedOutputs[i] = !op->get_output_target_inputs(i).empty();
+    }
+
+    sorted = ov::as_type_ptr<ov::op::v10::Unique>(op)->get_sorted();
+    if (op->get_input_size() > AXIS) {
+        flattened = false;
+        axis = ov::as_type<ov::op::v0::Constant>(op->get_input_node_ptr(AXIS))->cast_vector<int>()[0];
+        if (axis < 0) {
+            axis += op->get_input_partial_shape(IN_DATA).rank().get_length();
+        }
+        if (axis < 0 || axis >= op->get_input_partial_shape(IN_DATA).rank().get_length()) {
+            THROW_ERROR << "has invalid axis value: " << ov::as_type<ov::op::v0::Constant>(op->get_input_node_ptr(AXIS))->cast_vector<int>()[0];
+        }
+    } else {
+        flattened = true;
+    }
+}
+
+void Unique::initSupportedPrimitiveDescriptors() {
+    dataPrecision = getOriginalInputPrecisionAtPort(IN_DATA);
+    if (dataPrecision != Precision::I32 && dataPrecision != Precision::I8 && dataPrecision != Precision::U8) {
+        dataPrecision = Precision::FP32;
+    }
+    dataTypeSize = dataPrecision.size();
+    const InferenceEngine::Precision axisPrecision = Precision::I32;
+
+    impl_desc_type implType = ref;
+
+    std::vector<PortConfigurator> inPortConfigs = { {LayoutType::ncsp, dataPrecision} };
+    if (!flattened) {
+        inPortConfigs.push_back({LayoutType::ncsp, axisPrecision});
+    }
+    std::vector<PortConfigurator> outPortConfigs;
+    for (int i = 0; i < 4; i++) {
+        outPortConfigs.push_back({LayoutType::ncsp, i == 0 ? dataPrecision : axisPrecision});
+    }
+
+    addSupportedPrimDesc(inPortConfigs, outPortConfigs, implType, isDynamicNode());
+}
+
+void Unique::createPrimitive() {
+    Node::createPrimitive();
+}
+
+void Unique::prepareParams() {
+    auto& dataMemPtr = getParentEdgeAt(IN_DATA)->getMemoryPtr();
+    if (!dataMemPtr || !dataMemPtr->isAllocated()) {
+        THROW_ERROR << " has not allocated input data memory.";
+    }
+    for (int i = 0; i < 4; i++) {
+        if (definedOutputs[i]) {
+            auto& dstMemPtr = getChildEdgeAt(i)->getMemoryPtr();
+            if (!dstMemPtr || !dstMemPtr->isAllocated()) {
+                THROW_ERROR << " has not allocated output memory at port " << i;
+            }
+        }
+    }
+    if (getSelectedPrimitiveDescriptor() == nullptr) {
+        THROW_ERROR << " has unidentified preferable primitive descriptor.";
+    }
+}
+
+void Unique::execute(dnnl::stream strm) {
+    if (flattened) {
+        size_t uniqueLen = 1lu;
+        if (dataPrecision == Precision::FP32) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::FP32>::value_type>();
```

Comment:
Please support conditional compilation feature both for `flattenTensorExec` and `slicedTensorExec` via OV_SWITCH

Reply:
Done